docs/howto/custom-operator.rst

```diff
@@ -0,0 +1,192 @@
+                super().__init__(*args, **kwargs)
+                self.name = name
+                self.mysql_conn_id = mysql_conn_id
+                self.database = database
+
+            def execute(self, context):
+                hook = MySqlHook(mysql_conn_id=self.mysql_conn_id,
+                         schema=self.database)
+                sql = "select name from user"
+                result = hook.get_first(sql)
+                message = "Hello {}".format(result['name'])
+                print(message)
+                return message
+
+When the operator invokes the query on the hook object, a new connection gets created if it doesn't exist. 
+The hook retrieves the auth parameters such as username and password from Airflow
+backend and passes the params to the :py:func:`airflow.hooks.base_hook.BaseHook.get_connection`. 
+You should create hook only in the ``execute`` method or any method which is called from ``execute``.
+The constructor gets called whenever Airflow parses a DAG which happens frequently.
+The ``execute`` gets called only during a DAG run.
+
+User interface
+^^^^^^^^^^^^^^^
+Airflow also allows the developer to control how the operator shows up in the DAG UI.
+Override ``ui_color`` to change the background color of the operator in UI. 
+Override ``ui_fgcolor`` to change the color of the label.
+
+.. code::  python
+
+        class HelloOperator(BaseOperator):
+            ui_color = '#ff0000'
+            ui_fgcolor = '#000000'
+            ....
+
+Templating
+^^^^^^^^^^^
+You can use :ref:`Jinja templates <jinja-templating>` to parameterize your operator.
+Airflow considers the field names present in ``template_fields``  for templating while rendering
+the operator.
+
+.. code:: python
+    
+        class HelloOperator(BaseOperator):
+            
+            template_fields = ['name']
+            
+            @apply_defaults
+            def __init__(
+                    self,
+                    name: str,
+                    *args, **kwargs) -> None:
+                super().__init__(*args, **kwargs)
+                self.name = name
+
+            def execute(self, context):
+                message = "Hello from {}".format(name)
+                print(message)
+                return message
+
+You can use the template as follows:
+
+.. code:: python
+
+        with dag:
+            hello_task = HelloOperator(task_id='task_id_1', dag=dag, name='{{ task_id }}')
+
+In this example, Jinja looks for the ``name`` parameter and substitutes ``{{ task_id }}`` with
+``task_id_1``.
+
+
+The parameter can also contain a file name, for example, a bash script or a SQL file. You need to add
+the extension of your file in ``template_ext``. If a ``template_field`` contains a string ending with
+the extension mentioned in ``template_ext``, Jinja reads the content of the file and replace the templates
+with actual value.Note that Jinja substitutes the operator attributes and not the args. 
```

Comment:
```suggestion
with actual value. Note that Jinja substitutes the attributes on the instance, and not the args passed to
the constructor so make sure that `template_fields` contains the attribute names where you store the
values. 

For example this would not work as you might intend:

.. code:: python

        class HelloOperator(BaseOperator):
            
            template_fields = ['name']
            
            @apply_defaults
            def __init__(
                    self,
                    name: str,
                    *args, **kwargs) -> None:
                super().__init__(*args, **kwargs)
                self._name = name

(It would fail with an ``AttributeError`` exception raised.) In this case you would need to
have ``template_fields = ['_name']``, or to have ``self.name = name``.
```

Though I'm not sure if we want a positive example (do it this way) or a negative one (don't do it this way)

Reply:
.... read ahead Ash. It's right there. (Sorry.)